.github/workflows/test.yml

```diff
@@ -0,0 +1,28 @@
+name: test
+
+on:
+  push:
+#    branches:
+#      - main
+  pull_request:
+#    branches:
+#      - main
```

Comment:
We can leave this as is for now or re-enable for pushes to main and pull requests against main before merging.

Reply:
I think a good start for this would to have it run on pull requests against main before merging.
Even thought you said it takes a few minutes for these to run, it can and will provide good value.
And I'm sure that's something that can be sped up eventually.

Reply:
Sorry, my comment was a bit confusing. Currently with these lines commented out, it will run for every push and every pull request. I had to do this in order for the tests to run in this pull request (before being merged into `main`). The alternative would be to limit these to running only on push to `main` and pull request into `main`.

Reply:
Either way, uncommented for now.

Reply:
Also, looks like tests are running on this pull request even after uncommenting and not sure why I needed to comment out the first time. Maybe a momentary github actions issue.

Reply:
sounds good!